x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPutTransformAction.java

```diff
@@ -208,10 +213,20 @@ protected void masterOperation(Task task, Request request, ClusterState clusterS
             );
             return;
         }
-        try {
-            SourceDestValidator.validate(config, clusterState, indexNameExpressionResolver, request.isDeferValidation());
-        } catch (ElasticsearchStatusException ex) {
-            listener.onFailure(ex);
+
+        ValidationException validationResult = SourceDestValidator.validate(
+            clusterState,
+            indexNameExpressionResolver,
+            transportService.getRemoteClusterService(),
+            isRemoteSearchEnabled ? new RemoteClusterLicenseChecker(client, XPackLicenseState::isTransformAllowedForOperationMode) : null,
+            config.getSource().getIndex(),
+            config.getDestination().getIndex(),
+            clusterService.getNodeName(),
+            "basic",
```

Comment:
```suggestion
            License.OperationMode.BASIC.description()
```
We should use the operation mode enum.